Nuke EKS clusters

yorinasub17 opened the pull request:
Given how expensive EKS is (one cluster = $0.20/hr), and how frequent the tests fail right now as I actively develop, I am afraid I might forget to clean up one of these so decided to go ahead and implement the nuking.

Note that this required upgrading `aws-sdk-go`. I tried the latest and there were issues, so downgraded to what we have in terratest.

brikis98 commented on aws/eks_utils_for_test.go:
```diff
@@ -0,0 +1,106 @@
+	}
+	rand.Seed(time.Now().UnixNano())
+	randIndex := rand.Intn(len(supportedRegions))
+	return supportedRegions[randIndex]
```
There's a terratest `GetRandomRegion` helper that takes in a whitelist and/or blacklist of regions you can use...

brikis98 commented on aws/eks.go:
```diff
@@ -0,0 +1,95 @@
+)
+
+// getAllEksClusters returns a list of strings of EKS Cluster Names that uniquely identify each cluster.
+func getAllEksClusters(awsSession *session.Session, excludeAfter time.Time) ([]*string, error) {
```
Don't you need to call this EKS code from somewhere? I think there's a struct of some sort you need to add and then some extra API calls in the main CLI app...

yorinasub17 replied:
Yup... I can't believe I missed that...

yorinasub17 replied:
I was so tied up in the tests and it was getting late that I got a bit too trigger happy I guess.

yorinasub17 commented:
One downside of this: the tests now have an extra 20 minutes because launching and tearing down EKS clusters take that long...

brikis98 commented:
> One downside of this: the tests now have an extra 20 minutes because launching and tearing down EKS clusters take that long...

Does it run in parallel with the other tests so the whole build takes 20 min? Or is it sequential?

yorinasub17 commented:
It runs parallel, so the whole build takes 20 minutes. For comparison, it was ~5 mins before.

brikis98 commented:
Understood. Not much we can do, TBH. Eventually, we'll want to nuke RDS, Elasticsearch, and other slow stuff too, so the build will only get slower. Such is DevOps.

rfvermut commented:
Can anything be done to finish this?

autero1 commented:
We don't seem to have verbose output in the tests. Are there any particular tests that keep failing?

tonerdo commented:
@autero1 see #50

yorinasub17 commented:
Thanks @tonerdo for the test fixes! I took a quick scan and I believe I addressed all the comments, so I am going to go ahead and merge this and release!
